autoPyTorch/datasets/resampling_strategy.py

```diff
@@ -176,10 +101,96 @@ def time_series_cross_validation(num_splits: int, indices: np.ndarray, **kwargs:
         return splits
 
-    @classmethod
-    def get_cross_validators(cls, *cross_val_types: CrossValTypes) -> Dict[str, CrossValFunc]:
-        cross_validators = {
-            cross_val_type.name: getattr(cls, cross_val_type.name)
-            for cross_val_type in cross_val_types
-        }
-        return cross_validators
+
+class CrossValTypes(Enum):
+    """The type of cross validation
+
+    This class is used to specify the cross validation function
+    and is not supposed to be instantiated.
+
+    Examples: This class is supposed to be used as follows
+    >>> cv_type = CrossValTypes.k_fold_cross_validation
+    >>> print(cv_type.name)
+
+    k_fold_cross_validation
+
+    >>> print(cv_type.value)
+
+    functools.partial(<function CrossValTypes.k_fold_cross_validation at ...>)
+
+    >>> for cross_val_type in CrossValTypes:
+            print(cross_val_type.name)
+
+    stratified_k_fold_cross_validation
+    k_fold_cross_validation
+    stratified_shuffle_split_cross_validation
+    shuffle_split_cross_validation
+    time_series_cross_validation
+
+    Additionally, CrossValTypes.<function> can be called directly.
+    """
+    stratified_k_fold_cross_validation = partial(CrossValFuncs.stratified_k_fold_cross_validation)
+    k_fold_cross_validation = partial(CrossValFuncs.k_fold_cross_validation)
+    stratified_shuffle_split_cross_validation = partial(CrossValFuncs.stratified_shuffle_split_cross_validation)
+    shuffle_split_cross_validation = partial(CrossValFuncs.shuffle_split_cross_validation)
+    time_series_cross_validation = partial(CrossValFuncs.time_series_cross_validation)
+
+    def is_stratified(self) -> bool:
+        stratified = [self.stratified_k_fold_cross_validation,
+                      self.stratified_shuffle_split_cross_validation]
+        return getattr(self, self.name) in stratified
+
+    def __call__(self, num_splits: int, indices: np.ndarray, stratify: Optional[Any]
+                 ) -> Tuple[np.ndarray, np.ndarray]:
+        """TODO: doc-string and test files"""
+        self.value(num_splits=num_splits, indices=indices, stratify=stratify)
+
+    @staticmethod
+    def get_validators(*choices: CrossValFunc):
+        """TODO: to be compatible, it is here now, but will be deprecated soon."""
+        return {choice.name: choice.value for choice in choices}
+
+
+class HoldoutValTypes(Enum):
+    """The type of hold out validation (refer to CrossValTypes' doc-string)"""
+    holdout_validation = partial(HoldoutValFuncs.holdout_validation)
+    stratified_holdout_validation = partial(HoldoutValFuncs.stratified_holdout_validation)
+
+    def is_stratified(self) -> bool:
+        stratified = [self.stratified_holdout_validation]
+        return getattr(self, self.name) in stratified
+
+    def __call__(self, val_share: float, indices: np.ndarray, stratify: Optional[Any]
+                 ) -> Tuple[np.ndarray, np.ndarray]:
+        self.value(val_share=val_share, indices=indices, stratify=stratify)
```

Comment:
Now we can call the function directly in a way that `HoldoutValTypes.holdout_validation()`.